pennylane/operation.py

```diff
@@ -462,29 +375,136 @@ def queue(self):
         return self  # so pre-constructed Observable instances can be queued and returned in a single statement
```

Comment:
I see `Operator.queue` no longer checks that there is a valid `QNode` context (and gives an error message if there isn't), instead `Operator.__init__` sets do_queue to False if there is no context so `Operator.queue` isn't called at all.

While this makes writing tests faster since there is no longer a reason to set do_queue=False, it also will cause the system to fail silently if the user misunderstands how QNodes/Operators/quantum functions work and cause some puzzling bugs for the end user, which is why I'd consider going back to the way things used to work.

The purpose of the do_queue flag is to force the user to go through some extra trouble if they want to get the exceptional behavior and not have the Operator be queued into a QNode.

Reply:
Note that this change was made in #359 by @co9olguy, and is showing up here solely due to a merge from master.

Reply:
Yeah, I just noticed it.

Reply:
But I'm still concerned about the silent failure that can happen. For the tests a cleaner solution would IMO be to monkeypatch `Operation.queue` to do nothing.

Reply:
But that's another PR, this one looks ok.